### PR TITLE
Start building a Cycloid Drawer

### PR DESCRIPTION
**Pull request by peterjmartinson**
DON'T APPROVE THIS YET!!
Why doesn't it work?

1. Get a circle to apparently roll back and forth
2. Get it to draw a cycloid (but not)

I can't tell what the problem is!  Trying to get the cycloid to be drawn from an array of positions, but JS is freaking out about trying to add points.

### CODE REVIEW

File: src/features/draw-rolling-circle/Drawing.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    if (rollingCircle.centerX >= WIDTH - radius || rollingCircle.centerX <= radius) {
+      rollingCircle.direction *= -1
+    }
+
+    ctx.clearRect(0, 0, WIDTH, HEIGHT)
+    ctx.save()
+
+    // fill in background, just for reference
+    ctx.fillStyle = Color.BLACK
```
Comment (belcurv) [inline src/features/draw-rolling-circle/Drawing.js:54]: You'll need to add `BLACK` to `/config.colors.js`

File: src/features/draw-rolling-circle/Drawing.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    ctx.strokeStyle = Color.RED
+    ctx.stroke()
+
+    // draw the cycloid
+    ctx.beginPath()
+    ctx.moveTo(cycloid.x[0], cycloid.y[0])
+    for (let i = 1; i < cycloid.i; i++) {
+      ctx.lineTo(cycloid.x[i], cycloid.y[i])
+    }
+    ctx.strokeStyle = Color.Yellow
```
Comment (belcurv) [inline src/features/draw-rolling-circle/Drawing.js:84]:
```suggestion
    ctx.strokeStyle = Color.YELLOW
```

File: src/features/draw-rolling-circle/Drawing.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    // text box
+    ctx.fillText(`cycloid.x:  ${cycloid.x}`, 5, 10)
+
+    ctx.restore()
+    if (!isRollingCirclePaused) {
+      rollingCircle.theta += rollingCircle.speed * rollingCircle.direction
+      rollingCircle.centerX += rollingCircle.speed * radius * rollingCircle.direction
+      cycloid.i++
+      cycloid.x.push(pointX)
+      cycloid.Y.push(pointY)
```
Comment (belcurv) [inline src/features/draw-rolling-circle/Drawing.js:96]: Lowercase `y`

```suggestion
      cycloid.y.push(pointY)
```

File: src/features/draw-rolling-circle/Drawing.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
+// @ts-check
+
+import { CanvasBuilder } from '@/components/Canvas'
+import { Color, Physics } from '@/config'
+
+const HEIGHT = 100
+const WIDTH = 500
+const radius = 30
+let pointX, pointY
```
Comment (belcurv) [inline src/features/draw-rolling-circle/Drawing.js:9]: Based on the code so far, I don't think we need to store these outside of the draw factory's / draw function's scope.

Down inside `draw`, you can 

```js
    // draw the line
    const pointX = rollingCircle.centerX + radius * Math.cos(rollingCircle.theta)
    const pointY = rollingCircle.centerY + radius * Math.sin(rollingCircle.theta)
```

File: src/features/draw-rolling-circle/Drawing.js (hunk 1 above)
Comment (peterjmartinson) [reply src/features/draw-rolling-circle/Drawing.js:54]: Funny - it works!  I'll do this though.

File: src/features/draw-rolling-circle/Drawing.js (hunk 2 above)
Comment (peterjmartinson) [reply src/features/draw-rolling-circle/Drawing.js:84]: Same - it works!

File: src/features/draw-rolling-circle/Drawing.js (hunk 3 above)
Comment (peterjmartinson) [reply src/features/draw-rolling-circle/Drawing.js:96]: that's it.  Good eye.

File: src/features/draw-rolling-circle/Drawing.js (hunk 2 above)
Comment (peterjmartinson) [reply src/features/draw-rolling-circle/Drawing.js]: Looks like Canvas ignores this, and just retains the last good strokeStyle setting, which is RED in this case.

File: src/features/draw-rolling-circle/Drawing.js (hunk 4 above)
Comment (peterjmartinson) [reply src/features/draw-rolling-circle/Drawing.js]: Cleaned up.

**Review by belcurv — APPROVED** [review]

**Pull request merged**